src/database/index.tsx

```diff
@@ -390,27 +392,36 @@ export const getUser = async (auth0Id: string): Promise<User | null> => {
 
 export const getScenarioUsers = async (
   scenarioId: string,
-): Promise<Array<User> | null> => {
+): Promise<ScenarioUsers | null> => {
   try {
     const scenario = await getScenario(scenarioId);
 
     if (!scenario) {
       throw new Error(`No scenario found matching id ${scenarioId}`);
     }
 
-    const auth0Ids = Object.keys(scenario.roles);
+    const ownerId = findKey(scenario.roles, (role) => role === "owner");
+    const viewerIds = Object.entries(scenario.roles)
+      .filter(([, role]) => role === "viewer")
+      .map(([id]) => id);
 
     // We have to get the users one at a time because Firestore limits the number
     // of elements that can be included within an "in" clause to 10. In other
     // words, if a Scenario has more than 10 users, writing the query with a
     // where clause that includes each of these user's auth0Ids would fail.
     // https://firebase.google.com/docs/firestore/query-data/queries#in_and_array-contains-any
-    const users = await Promise.all(
-      auth0Ids.map((auth0Id) => getUser(auth0Id)),
-    );
-
-    // Appeasing the TypeScript gods by filtering out possible null values
-    return users.filter((user): user is User => user !== null);
+    const userFetches = viewerIds.map((auth0Id) => getUser(auth0Id));
+    if (ownerId) userFetches.push(getUser(ownerId));
```

Comment:
Would there ever not be an `ownerId`?

Reply:
in practice no, but according to TypeScript yes

Reply:
actually this wasn't a typescript thing it was just me being overly cautious about the schema-less database ... there is nothing anywhere that actually requires there to be an owner, we just assume there always will be one. this particular function would work perfectly well without one, should the situation arise, so I decided to just handle it?